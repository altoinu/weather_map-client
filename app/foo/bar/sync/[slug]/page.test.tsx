import Page from "./page";
import { act, render, screen } from "@testing-library/react";

describe("slug page", () => {
  it("should render", async () => {
    const { container: pageContainer } = await act(async () => {
      return render(
        <Page
          params={Promise.resolve({
            slug: "hello!",
          })}
        />,
      );
    });

    expect(pageContainer).toMatchSnapshot();
  });

  it("should render a heading", async () => {
    await act(async () => {
      return render(
        <Page
          params={Promise.resolve({
            slug: "hello!",
          })}
        />,
      );
    });

    const heading = screen.getByRole("heading");
    expect(heading.textContent).toBe("Slug: hello!");
  });
});
