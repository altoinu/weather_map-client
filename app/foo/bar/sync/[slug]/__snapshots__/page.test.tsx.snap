// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`slug page should render 1`] = `
<div>
  <h3>
    Slug: 
    hello!
  </h3>
</div>
`;
